test/integration/scheduler/scheduler_test.go

```diff
@@ -647,3 +651,198 @@ func TestAllocatable(t *testing.T) {
 		t.Logf("Test allocatable awareness: %s Pod not scheduled as expected", testAllocPod2.Name)
 	}
 }
+
+// TestPreemptionBasic tests that basic preemption logic works.
+func TestPreemptionBasic(t *testing.T) {
+	// Enable PodPriority feature gate.
+	utilfeature.DefaultFeatureGate.Set(fmt.Sprintf("%s=true", features.PodPriority))
+	// Initialize scheduler.
+	context := initTest(t, "preemption")
+	defer cleanupTest(t, context)
+	// Create a node with some resources.
+	nodeRes := &v1.ResourceList{
+		v1.ResourcePods:   *resource.NewQuantity(32, resource.DecimalSI),
+		v1.ResourceCPU:    *resource.NewMilliQuantity(100, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(200, resource.BinarySI),
+	}
+	_, err := createNode(context.clientSet, "node1", nodeRes)
+	if err != nil {
+		t.Fatalf("Error creating nodes: %v", err)
+	}
+	// Run a low priority pod that uses all the node resources.
+	lowPriority, highPriority := int32(100), int32(1000)
+	victim, err := runPausePod(context.clientSet, initPausePod(context.clientSet, &pausePodConfig{
+		Name:      "victim-pod",
+		Namespace: context.ns.Name,
+		Priority:  &lowPriority,
+		Resources: &v1.ResourceRequirements{Requests: v1.ResourceList{
+			v1.ResourceCPU:    *resource.NewMilliQuantity(100, resource.DecimalSI),
+			v1.ResourceMemory: *resource.NewQuantity(200, resource.BinarySI)},
+		},
+	}))
+	if err != nil {
+		t.Fatalf("Error running pause pod: %v", err)
+	}
+	// Now run a higher priority pod and make sure that the pod is scheduled.
+	_, err = createPausePod(context.clientSet, &pausePodConfig{
+		Name:      "preemptor-pod",
+		Namespace: context.ns.Name,
+		Priority:  &highPriority,
+		Resources: &v1.ResourceRequirements{Requests: v1.ResourceList{
+			v1.ResourceCPU:    *resource.NewMilliQuantity(100, resource.DecimalSI),
+			v1.ResourceMemory: *resource.NewQuantity(200, resource.BinarySI)},
+		},
+	})
+	if err != nil {
+		t.Errorf("Error while creating high priority pod: %v", err)
+	}
+	// Check that the lower priority pod is preempted.
+	if err = wait.Poll(time.Second, 5*time.Second, podIsGettingEvicted(context.clientSet, victim.Namespace, victim.Name)); err != nil {
+		t.Errorf("Victim pod is not getting evicted.")
+	}
+	// Also check that the preemptor pod gets the annotation for nominated node name.
+	if err = wait.Poll(time.Second, 5*time.Second, func() (bool, error) {
```

Comment:
5 seconds is asking for flakes in integration tests when run in parallel under load